converters/toZigbee.js

```diff
@@ -1170,6 +1170,35 @@ const converters = {
             return converters.light_colortemp.convert(key, value, message, type, postfix);
         },
     },
+    cover_position: {
+        key: ['position'],
```

Comment:
This should be 'state' (as homeassistant sends `100`, not `{"position": 100}`, when the payload is not JSON, `state` will be used as default).

Reply:
I probably need more explanation on this ... :)

I was reading this https://www.home-assistant.io/components/cover.mqtt/#set_position_topic
and it sounds like I should use "set_position_topic" instead of "command_topic"

So in this case, I'd think it should be like this?

```
    cover_position: {
        key: ['state'],
        convert: (key, value, message, type, postfix) => {
            const cid = 'genLevelCtrl';
            const attrId = 'currentLevel';

            if (type === 'set') {
                value = Math.round(Number(value) * 2.55).toString();
                return {
                    cid: cid,
                    cmd: 'moveToLevelWithOnOff',
                    cmdType: 'functional',
                    zclData: {
                        level: value,
                        transtime: 0,
                    },
                    cfg: cfg.default,
                    readAfterWriteTime: 0,
                };
            } else if (type === 'get') {
                return {
                    cid: cid,
                    cmd: 'read',
                    cmdType: 'foundation',
                    zclData: [{attrId: zclId.attr(cid, attrId).value}],
                    cfg: cfg.default,
                };
            }
        },
    },
```

What I'm also confused about is the 'get' part, I assume it's used by "position_topic"? But as you can see the return value for get is just a command object, how do I convert from brightness to position (0-100) ?

Reply:
correction not brightness but currentLevel which is 0-255 based